synapse/config/logger.py

```diff
@@ -237,6 +220,26 @@ def factory(*args: Any, **kwargs: Any) -> logging.LogRecord:
 
     logging.setLogRecordFactory(factory)
 
+    # Configure the logger with the initial configuration.
```

Comment:
I don't think this *has* to be moved, but since our logging generally assumes that a `request` exists on the `Record` it makes sense to set that up *first*, in case something quickly logs.